### PR TITLE
Fix overlay popups event routing

### LINKED ISSUE #10420: ComboBox does not allow selection with OverlayPopups=true
**Describe the bug**
If .With(new Win32PlatformOptions { OverlayPopups = true}) is used you cannot select from the dropdown list of a ComboBox.

**To Reproduce**

Add this to a a window
```
<ComboBox>
	<ComboBoxItem>option 1</ComboBoxItem>
	<ComboBoxItem>option 2</ComboBoxItem>
</ComboBox>
```
and set .With(new Win32PlatformOptions { OverlayPopups = true}) in programs.cs appbuilder

**Expected behavior**
Clicking an option in the dropdown should select it

**Desktop (please complete the following information):**

- OS: windows
- Version 11.0.999-cibuild0030320-beta

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

As described in #10420, event routing for overlay popups was broken in #10191. The fix was simple, to use `Parent` instead of `VisualParent` as `OverlayPopupHost`'s interactive parent.

The rest of this PR is adding tests for overlay popups to the integration tests to try to catch this kind of problem early in future.

## Fixed issues

Fixes #10420

### CODE REVIEW

File: Avalonia.sln [hunk 1]
```diff
@@ -555,9 +555,14 @@ Global
 		{75C47156-C5D8-44BC-A5A7-E8657C2248D6}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{75C47156-C5D8-44BC-A5A7-E8657C2248D6}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{75C47156-C5D8-44BC-A5A7-E8657C2248D6}.Release|Any CPU.Build.0 = Release|Any CPU
+		{C810060E-3809-4B74-A125-F11533AF9C1B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (grokys) [inline Avalonia.sln:558]: Visual Studio didn't seem to like the `.sln` file as it was and kept changing it.

File: src/Avalonia.Controls/Primitives/OverlayPopupHost.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ bool IPopupHost.Topmost
         }
 
         /// <inheritdoc />
-        protected internal override Interactive? InteractiveParent => (Interactive?)VisualParent;
+        protected internal override Interactive? InteractiveParent => Parent as Interactive;
```
Comment (grokys) [inline src/Avalonia.Controls/Primitives/OverlayPopupHost.cs:54]: This is the actual fix.

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030515-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
@grokys seems like tests are failing

**Comment by grokys** [comment]
@maxkatz6 yeah it revealed a different problem with menus: fixed that in a [separate PR](https://github.com/AvaloniaUI/Avalonia/pull/10461) if CI passes there, merging that one will also merge this one.

**Pull request merged**